components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/DefaultLoginValidationDelegate.kt

```diff
@@ -84,13 +82,12 @@ class DefaultLoginValidationDelegate(
                 }
                 result
             } catch (e: InvalidRecordException) {
-                when (e.reason) {
-                    InvalidLoginReason.DUPLICATE_LOGIN -> Result.Error.Duplicate
-                    InvalidLoginReason.EMPTY_PASSWORD -> Result.Error.EmptyPassword
-                    InvalidLoginReason.EMPTY_ORIGIN -> Result.Error.GeckoError(e)
-                    InvalidLoginReason.BOTH_TARGETS -> Result.Error.GeckoError(e)
-                    InvalidLoginReason.NO_TARGET -> Result.Error.GeckoError(e)
-                    InvalidLoginReason.ILLEGAL_FIELD_VALUE -> Result.Error.GeckoError(e)
+                // This feels a bit hacky but unforunately a current limitation of uniffi and how it sends errors across ffi
+                // Potentially update after: https://github.com/mozilla/uniffi-rs/issues/460
+                when(e.message) {
+                    "InvalidLogin::DuplicateLogin" -> Result.Error.Duplicate
+                    "InvalidLogin::EmptyPassword" -> Result.Error.EmptyPassword
+                    else -> Result.Error.GeckoError(e)
```

Comment:
The original expressiveness had the benefit of guarding us from accidentally wrapping new exception types w/ `Result.Error.GeckoError`, in cases where it may not be appropriated. You could introduce a `Result.Error.Other(Exception)` for dealing with cases that aren't explicitly recognized.

Reply:
Hmmm, this actually made me think about this more and I _think_ the "right" way to adapt this is actually to add the rest of the enums found on the a-s side [InvalidLoginReason enum](https://github.com/mozilla/application-services/blob/72b827c3e0f883163762857fd766df1aeb060725/components/logins/src/error.rs#L66)  to the error class in [LoginsStorage.kt](https://github.com/mozilla-mobile/android-components/blob/bfaf514793dd20f9f12d3f551f9ff3e470f52727/components/concept/storage/src/main/java/mozilla/components/concept/storage/LoginsStorage.kt#L211). Which the above currently seems to partially match up to the on the a-s side of things.

@grigoryk would there be any issue with me essentially matching the `Result.Error` class to the a-s `InvalidLoginReason` enum?

Reply:
I think that should be fine, @skhamis!